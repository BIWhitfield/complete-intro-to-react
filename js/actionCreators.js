// @flow
import { SET_SEARCH_TERM, ADD_API_DATA } from './actions';

export function setSearchTerm(searchTerm: string) {
  return { type: SET_SEARCH_TERM, payload: searchTerm };
}

export function addAPIData(apiData: Show) {
  return { type: ADD_API_DATA, payload: apiData };
}
