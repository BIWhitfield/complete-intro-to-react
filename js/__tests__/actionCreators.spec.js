// @flow

import moxios from 'moxios';
import { setSearchTerm, addAPIData } from '../actionCreators';
import getAPIDetails from '../asyncActions';

const Atlanta = {
  title: 'Atlanta',
  year: '2008â€“2013',
  description: 'Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; "Earnest \'Earn\' Marks," an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.',
  poster: 'a.jpg',
  imdbID: 'tt4288182',
  trailer: 'MpEdJ-mmTlY',
  rating: '8.6'
};

// test action creators with snapshots
test('setSearchTerm', () => {
  expect(setSearchTerm('black')).toMatchSnapshot();
});

test('addAPIData', () => {
  expect(addAPIData(Atlanta)).toMatchSnapshot();
});

test('getAPIDetails', (done: Function) => {
  const dispatchMock = jest.fn();
  moxios.withMock(() => {
    getAPIDetails(Atlanta.imdbID)(dispatchMock);
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request
        .respondWith({
          status: 200,
          response: Atlanta
        })
        .then(() => {
          expect(request.url).toEqual(`http://localhost:3000/${Atlanta.imdbID}`);
          expect(dispatchMock).toBeCalledWith(addAPIData(Atlanta));
          done();
        });
    });
  });
});
