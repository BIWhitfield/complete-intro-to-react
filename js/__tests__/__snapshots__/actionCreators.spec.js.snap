// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addAPIData 1`] = `
Object {
  "payload": Object {
    "description": "Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; \\"Earnest 'Earn' Marks,\\" an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.",
    "imdbID": "tt4288182",
    "poster": "a.jpg",
    "rating": "8.6",
    "title": "Atlanta",
    "trailer": "MpEdJ-mmTlY",
    "year": "2008â€“2013",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`setSearchTerm 1`] = `
Object {
  "payload": "black",
  "type": "SET_SEARCH_TERM",
}
`;
